doc/architecture.md

```diff
@@ -56,32 +72,69 @@ The frontend application will subscribe to the EVM transaction log, enabling the
 
 ![Frontend High-Level Architecture](architecture/img/frontend-hl-arch.png)
 
+The Squid library showed in the above diagram encapsulates the logic to deal with the Ocean components (Keeper & Provider). Squid libraries in different languages are part of the Tier 2.
 
-### Keeper Smart Contracts
+#### Datascience Tools
 
-The Keeper Smart Contracts are Solidity Smart Contracts deployed and running in the decentralized [Ethereum Virtual Machine (EVM)](https://github.com/ethereum/wiki/wiki/Ethereum-introduction#about-ethereum).
+Datascience Tools are the interface to Ocean used by datascientists. Typically in Python (but not limited to it),
+those tools and libraries expose a high-level API allowing to integrate the Ocean capabilities in different kind of computation pipelines.
 
-The Ocean Keeper implementation is where we put the following modules together:
 
-- **TCRs** - Users create challenges and resolve them through voting to maintain registries.
-- **Ocean Tokens** - The intrinsic tokens circulated inside the Ocean Network, which is used in the voting within TCRs;
-- **Curated Proofs Market** - The core component where people can transact with each other and curate assets through staking with Ocean tokens.
+### Tier 2 - Protocol Layer
 
-![Contracts Architecture](architecture/img/contracts-structure.jpg)
+Includes all the high-level libraries to interact with Ocean Protocol, and the Enabler components complementing the Smart Contracts.
 
+#### Squid Libraries
 
-### Provider
+Squid is a High Level specification API abstracting the interaction with the most relevant Ocean Protocol components.
+It allows to use the Ocean capabilities without requiring to know the complexity of Smart Contracts or Metadata storage systems.
+
+Squid API as specification, can be implemented in different languages, initially [Javascript](https://github.com/oceanprotocol/squid-js) (used in Pleuston) and [Python](https://github.com/oceanprotocol/squid-py) (used in the Datascience tools).
+
+The complete specification of the [Squid API](architecture/squid.md) can be reviewed as part of this repository.
```

Comment:
The link to `architecture/squid.md` doesn't work. There is no file at https://github.com/oceanprotocol/dev-ocean/tree/master/doc/architecture/squid.md and this PR doesn't add one.